include/openenclave/internal/syscall_decls.h

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_INTERNAL_SYSCALL_DECLS_H
+#define _OE_INTERNAL_SYSCALL_DECLS_H
+
+#include <bits/syscall.h>
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+
+// For OE_SYS_ defines.
+// They are just used for asserting that they are equal to the corresponding
+// SYS_ ones.
+#if __x86_64__ || _M_X64
+#include <openenclave/internal/syscall/sys/bits/syscall_x86_64.h>
+#elif defined(__aarch64__)
+#include <openenclave/internal/syscall/sys/bits/syscall_aarch64.h>
+#else
+#error Unsupported architecture
+#endif
+
+OE_EXTERNC_BEGIN
+
+#define OE_SYSCALL_NAME(idx) oe##idx##_impl
+#define OE_CHECK_IDX(idx) OE_STATIC_ASSERT((idx + 1) > 0)
+
+#define OE_SYSCALL_DISPATCH(idx, ...) \
+    case OE_##idx:                    \
+        return OE_SYSCALL_NAME(_##idx)(__VA_ARGS__)
```

Comment:
```suggestion
#define OE_SYSCALL_NAME(index) oe##index##_impl
#define OE_CHECK_IDX(index) OE_STATIC_ASSERT((index+ 1) > 0)

#define OE_SYSCALL_DISPATCH(index, ...) \
    case OE_##index:                    \
        return OE_SYSCALL_NAME(_##index)(__VA_ARGS__)
```
and same for line 40 etc.

Also I'd recommend `OE_CHECK_IDX` -> `OE_CHECK_INDEX`

Reply:
Done.